src/index.test.js

```diff
@@ -136,23 +136,40 @@ describe('MagicUrl', () => {
       })
     })
 
-    test('should parse same url multiple times', () => {
+    test('should parse url instead of mail', () => {
+      // the url is part of an mail and both start at the same index
```

Comment:
Haven't even thought about those edge cases 😅 

Actually in that case I would expect to only get the mail and not the url, because as you correctly state "the url is part of an mail".

Reply:
I agree on the expected behaviour. So I've switch the priority of the url and mail matching.

My initial idea was to make the changes as small as possible. And the current behaviour is more like favouring url against mail.

Reply:
Nice, thanks for the changes!